keras/trainers/epoch_iterator.py

```diff
@@ -170,18 +169,13 @@ def enumerate_epoch(self, return_type="np"):
                 if self._insufficient_data:
                     break
 
-                if tf.available:
-                    errors = (StopIteration, tf.errors.OutOfRangeError)
```

Comment:
I wonder if not excepting `tf.errors.OutOfRangeError` will have unforeseen consequences. Maybe we could catch everything, check the exception class, and warn, or reraise otherwise?

Or I guess that's not needed -- looking at the TF codebase this is virtually never raised

Reply:
Oh. Forgot to mention this. I don't think this is ever called when using TF backend.
TF backend uses TFEpochIterator, and the base class method of enumerate_epoch is never called.